### PR TITLE
Implement Enhanced Storage Capabilities with `get` and `count` Methods

### PR DESCRIPTION
**Pull request by freakinlight**
## Overview
This pull request introduces two critical enhancements to both the `DBStorage` and `FileStorage` systems within our project. By implementing `get` and `count` methods, we enhance the robustness and functionality of our storage operations, enabling more efficient data retrieval and manipulation.

## Changes Made
- **`get` Method**: Added to both `DBStorage` and `FileStorage` classes. This method facilitates the retrieval of an object based on its class type and ID, streamlining the process of data access across our application.
- **`count` Method**: Also added to both storage classes. This method provides the functionality to count the number of objects stored, either globally or within a specified class, enhancing our ability to manage and audit stored data effectively.

## Why These Changes?
The additions of `get` and `count` are pivotal for several reasons:
1. **Improved Efficiency**: Simplifies the process of accessing and counting objects in storage without the need for additional queries or operations.
2. **Enhanced Functionality**: These methods support more complex operations and integrations, allowing our application to scale more effectively and handle more dynamic data interactions.
3. **Better Testing and Maintenance**: Accompanying unit tests ensure these methods are robust and perform as expected, reducing bugs and increasing reliability.

## Additional Information
- The `get` method helps in directly fetching an object by ID, reducing the overhead of filtering through objects manually.
- The `count` method provides a quick overview of storage utilization, crucial for monitoring and analytics.

I encourage all reviewers to pull this branch, test the changes locally, and provide feedback or approve the pull request if everything looks good. These enhancements are a step towards optimizing our backend storage capabilities and making our data handling processes more streamlined and effective.

## Next Steps
Pending review, the next steps will involve integrating these methods into our daily operations and possibly extending their capabilities to support more complex queries and data interactions.

Thank you for reviewing this pull request!

### CODE REVIEW

File: console.py [hunk 1]
```diff
@@ -46,10 +46,10 @@ def _key_value_parser(self, args):
                 else:
                     try:
                         value = int(value)
-                    except:
+                    except Exception:
```
Comment (fuzzypixelz) [inline console.py]:
```suggestion
                    except ValueError:
```

File: console.py [hunk 2]
```diff
@@ -46,10 +46,10 @@ def _key_value_parser(self, args):
...
                         try:
                             value = float(value)
-                        except:
+                        except Exception:
```
Comment (fuzzypixelz) [inline console.py:52]:
```suggestion
                        except ValueError:
```

File: console.py [hunk 3]
```diff
@@ -140,12 +140,12 @@ def do_update(self, arg):
                                 if args[2] in integers:
                                     try:
                                         args[3] = int(args[3])
-                                    except:
+                                    except Exception:
```
Comment (fuzzypixelz) [inline console.py]:
```suggestion
                                    except ValueError:
```

File: console.py [hunk 4]
```diff
@@ -140,12 +140,12 @@ def do_update(self, arg):
...
                                         args[3] = 0
                                 elif args[2] in floats:
                                     try:
                                         args[3] = float(args[3])
-                                    except:
+                                    except Exception:
```
Comment (fuzzypixelz) [inline console.py:148]:
```suggestion
                                    except ValueError:
```

File: file.json
Comment (fuzzypixelz) [inline file.json:1]: Should this file really be committed? If so, give it a name that conveys its purpose better.

File: __pycache__/console.cpython-310.pyc
Comment (fuzzypixelz) [inline __pycache__/console.cpython-310.pyc:1]: Don't commit `__pycache__`.

File: models/__pycache__/__init__.cpython-310.pyc
Comment (fuzzypixelz) [inline models/__pycache__/__init__.cpython-310.pyc:1]: Don't commit `models/__pycache__`

File: models/engine/__pycache__/__init__.cpython-310.pyc
Comment (fuzzypixelz) [inline models/engine/__pycache__/__init__.cpython-310.pyc:1]: Don't commit `models/engine/__pycache__`.

File: models/engine/db_storage.py [hunk 5]
```diff
@@ -74,3 +74,16 @@ def reload(self):
     def close(self):
         """call remove() method on the private session attribute"""
         self.__session.remove()
+
+    def get(self, cls, id):
+        """Retrieve one object."""
+        if cls and id:
```
Comment (fuzzypixelz) [inline models/engine/db_storage.py]:
```suggestion
        if cls is not None and id is not None:
```

File: models/engine/db_storage.py [hunk 6]
```diff
@@ -74,3 +74,16 @@ def reload(self):
     def close(self):
         """call remove() method on the private session attribute"""
         self.__session.remove()
+
+    def get(self, cls, id):
```
Comment (fuzzypixelz) [inline models/engine/db_storage.py:92]:
```suggestion
    def get(self, cls=None, id=None):
```

File: models/engine/file_storage.py [hunk 7]
```diff
@@ -55,7 +55,7 @@ def reload(self):
                 jo = json.load(f)
             for key in jo:
                 self.__objects[key] = classes[jo[key]["__class__"]](**jo[key])
-        except:
+        except Exception:
```
Comment (fuzzypixelz) [inline models/engine/file_storage.py]:
```suggestion
        except OSError:
```

File: models/engine/file_storage.py [hunk 8]
```diff
@@ -68,3 +68,14 @@ def delete(self, obj=None):
     def close(self):
         """call reload() method for deserializing the JSON file to objects"""
         self.reload()
+
+    def get(self, cls, id):
+        """Retrieve one object."""
+        if cls and id:
```
Comment (fuzzypixelz) [inline models/engine/file_storage.py]:
```suggestion
        if cls is not None and id is not None:
```

File: models/engine/file_storage.py [hunk 9]
```diff
@@ -68,3 +68,14 @@ def delete(self, obj=None):
     def close(self):
         """call reload() method for deserializing the JSON file to objects"""
         self.reload()
+
+    def get(self, cls, id):
```
Comment (fuzzypixelz) [inline models/engine/file_storage.py:75]:
```suggestion
    def get(self, cls=None, id=None):
```

File: tests/__pycache__/__init__.cpython-310.pyc
Comment (fuzzypixelz) [inline tests/__pycache__/__init__.cpython-310.pyc:1]: Again, don't commit `tests/__pycache__`.

File: test_get_count.py
Comment (fuzzypixelz) [inline test_get_count.py:1]: There is a `tests` package, try to integrate this there.

File: tests/test_models/__pycache__/__init__.cpython-310.pyc
Comment (fuzzypixelz) [inline tests/test_models/__pycache__/__init__.cpython-310.pyc:1]: Again, don't commit `tests/test_models/__pycache__`.

**Review by fuzzypixelz — CHANGES_REQUESTED** [review]

**Review by EngineeringProwess — APPROVED** [review]

**Pull request closed without merging**